src/typing.py

```diff
@@ -1859,6 +1860,21 @@ def __new__(cls, *args, **kwds):
                             "create a subclass instead")
         return _generic_new(_G_base, cls, *args, **kwds)
 
+if hasattr(collections_abc, 'AsyncGenerator'):
+    _AG_base = collections_abc.AsyncGenerator
+
+    class AsyncGenerator(AsyncIterator[T_co], Generic[T_co, T_contra],
+                         extra=_AG_base):
+        __slots__ = ()
+
+        def __new__(cls, *args, **kwds):
```

Comment:
This was mostly copied from the `Generator` implementation just above. It is possible to instantiate a subclass of `AsyncGenerator`, which would use this `__new__`.